### DISCUSSION TITLE
branch for new release

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#1966] [General]
How about creating a branch for each new release?  Then it would be possible to fix major bugs in the latest release by cherry picking the fixes from main branch.

For example, now the latest (tagged) release is v2.5.0 and it contains a bug where user does not hear early audio after 183 response.  So it should be be possible to create a new release v2.5.1, where this fix (and possible other bug fixes in main) are included.

**Comment by sreimers** [comment]
I think a single stable branch should enough (had the same for re), since we only support the latest release. I have just released v2.5.1 this way.

**Comment by juha-h** [comment]
Sebastian Reimers writes:

> I think a single stable branch should enough (had the same for re),
> since we only support the latest release. I have just released v2.5.1
> this way.

OK, thanks.

**Comment by juha-h** [comment]
I tried to build 2.5.1 on Debian and noticed that `debian/control` is out of date regarding libre at least.

Also, if I'm on 2.5.1 branch and installed libre is latest main, build fails like this:
```
make[1]: Entering directory '/usr/src/orig/baresip'
  CC      build-x86_64/src/account.o
  CC      build-x86_64/src/aucodec.o
  CC      build-x86_64/src/audio.o
src/audio.c: In function ‘audio_alloc’:
src/audio.c:1200:9: warning: implicit declaration of function ‘mtx_alloc’; did you mean ‘mutex_alloc’? [-Wimplicit-function-declaration]
 1200 |  err  = mtx_alloc(&tx->mtx);
      |         ^~~~~~~~~
      |         mutex_alloc
...

**Comment by juha-h** [comment]
It is a bit confusing that main Makefile has 2.5.0.  After all this hassle, I still start to feel that it would be much cleaner to have branches.  If so, the current stable branch would be v2.5 with tags v2.5.0, v2.5.1, etc., and main (development branch) would have version 2.6.0 or 3.0.0. This is how many other open source projects operate.

**Reply by sreimers** [reply]
re/rem/retest/baresip have a stable branch now. We can bump main dev version to next major version 3.0.0 (or like other projects use even and uneven numbers) and create release tags like 2.6.0 on the stable branch (after merging main to stable every month).  I will update our [release checklist](https://github.com/baresip/re/wiki/Release-Checklist) @alfredh do you agree?

**Comment by alfredh** [comment]
I would prefer to stay on version 2.x.x for a while

example monthly releases:

* 2.6.0
* 2.7.0
* 2.8.0
* 2.9.0
* ... etc ...

on the other things I agree
